internal/db/schema/state.go

```diff
@@ -0,0 +1,57 @@
+package schema
+
+import (
+	"context"
+)
+
+const nilVersion = -1
+
+type migrationState struct {
```

Comment:
would you mind adding a godoc here about this type?  I know, it's not exported but it would be great.

Reply:
the doc made me chuckle... "an interface that..."  Don't leave me hanging like that :)